### ISSUE TITLE
Changing model_fn for existing model

### ISSUE THREAD

**Issue by ctcml** [aws/sagemaker-python-sdk#728]
Hi,

I have trained a PyTorch model on SageMaker on a GPU.
The model_fn function was written to deploy the model on a GPU machine as well.
However, I would like to deploy on a CPU machine to save cost, is there a way to update model_fn without retraining?

Specifically, I am changing 
`model.load_state_dict(torch.load(f))`
to
`model.load_state_dict(torch.load(f, map_location=device)))`

Any help or link to the docs regarding this is very much appreciated.

**Comment by icywang86rui** [comment]
Hi @ctcml,

How is the ``model_fn`` implemented is completely up to you. ``model_fn`` is only used during serving even if you have defined it in your training script as well. So if you have a trained SageMaker mode and want to serve it using a different script you can follow the instructions here - https://github.com/aws/sagemaker-python-sdk/tree/master/src/sagemaker/pytorch#deploying-endpoints-from-model-data

In the ``transform_script.py`` you can implement your ``model_fn`` with ``model.load_state_dict(torch.load(f, map_location=device)))``

Hope that answers your question.

**Comment by laurenyu** [comment]
closing due to inactivity. feel free to reopen if necessary!

**Comment by faizanahemad** [comment]
A question about `model_fn`, does `model_fn` run every-time a API call for inference comes (invoke_endpoint) or it runs only once when sagemaker starts the instance? 

I notice that logs in `model_fn` show up in cloudwatch after every inference call.

**Comment by laurenyu** [comment]
@faizanahemad could you open a new issue in this repo with your question? (it'll help with our internal tracking and it's a bit different from the original question here)
